src/Evolution/DiscontinuousGalerkin/InboxTags.hpp

```diff
@@ -109,19 +113,47 @@ struct BoundaryCorrectionAndGhostCellsInbox {
 
  public:
   using temporal_id = TimeStepId;
-  using type = std::map<TimeStepId, DirectionalIdMap<Dim, stored_type>>;
+  // Used by array implementation
+  using type_map = std::map<TimeStepId, DirectionalIdMap<Dim, stored_type>>;
 
-  template <typename Inbox, typename ReceiveDataType>
-  static void insert_into_inbox(const gsl::not_null<Inbox*> inbox,
-                                const temporal_id& time_step_id,
-                                ReceiveDataType&& data) {
-    auto& current_inbox = (*inbox)[time_step_id];
-    auto& [volume_mesh_of_ghost_cell_data, face_mesh, ghost_cell_data,
-           boundary_data, boundary_data_validity_range, boundary_tci_status] =
-        data.second;
-    (void)ghost_cell_data;
+  // Used by nodegroup implementation
+  using type_spsc = evolution::dg::AtomicInboxBoundaryData<Dim>;
+
+  // The actual type being used.
+  using type = type_map;
 
+  template <typename ReceiveDataType>
+  static size_t insert_into_inbox(const gsl::not_null<type_spsc*> inbox,
```

Comment:
You've changed the return type of this function. Could you add some docs about what the `size_t` represents? From the looks of it, it's just the new size of the inbox